executor/slow_query.go

```diff
@@ -118,9 +125,25 @@ func (e *slowQueryRetriever) close() error {
 	return nil
 }
 
-func (e *slowQueryRetriever) dataForSlowLog(ctx sessionctx.Context) ([][]types.Datum, error) {
+type parsedSlowLog struct {
+	rows [][]types.Datum
+	err  error
+}
+
+func (e *slowQueryRetriever) parseDataForSlowLog(ctx sessionctx.Context) {
 	reader := bufio.NewReader(e.files[e.fileIdx].file)
 	rows, err := e.parseSlowLog(ctx, reader, 1024)
+	e.parsedSlowLogCh <- parsedSlowLog{rows, err}
```

Comment:
Need loop here, `parseSlowLog` only return 1024 rows every time.

And need to check `ctx.Done()`

Reply:
Done